ui/src/main/java/com/hyperwallet/android/ui/view/widget/DateUtil.java

```diff
@@ -0,0 +1,133 @@
+     *
+     * @param serverDate Date from server
+     * @return String date in the format (dd-MMM-yyyy)
+     */
+    @NonNull
+    String convertDateFromServerToWidgetFormat(@Nullable final String serverDate) {
+        if (isServerDateNotMatched(serverDate)) {
+            return "";
+        }
+        try {
+            final Calendar calendar = Calendar.getInstance();
+            calendar.setTime(mServerDateFormat.parse(serverDate));
+            return mWidgetDateFormat.format(calendar.getTime());
+        } catch (NumberFormatException | ParseException | IndexOutOfBoundsException e) {
+            Log.e(TAG, e.getMessage());
+            return "";
+        }
+    }
+
+    /**
+     * Build Date in the server format (yyyy-MM-dd)
+     *
+     * @param year       the selected year
+     * @param month      the selected month (0-11 for compatibility with {@link Calendar#MONTH})
+     * @param dayOfMonth th selected day of the month (1-31, depending on month)
+     * @return String date in the format (yyyy-MM-dd)
+     */
+    @NonNull
+    String buildDateFromDateDialogToServerFormat(final int year, final int month, final int dayOfMonth) {
+        final Calendar calendar = getCalendar(year, month, dayOfMonth);
+        return mServerDateFormat.format(calendar.getTime());
+    }
+
+    /**
+     * Build Date in the DateWidget format (dd-MMM-yyyy)
+     *
+     * @param year       the selected year
+     * @param month      the selected month (0-11 for compatibility with {@link Calendar#MONTH})
+     * @param dayOfMonth th selected day of the month (1-31, depending on month)
+     * @return String date in the format (yyyy-MM-dd)
+     */
+    @NonNull
+    String buildDateFromDateDialogToWidgetFormat(final int year, final int month, final int dayOfMonth) {
+        final Calendar calendar = getCalendar(year, month, dayOfMonth);
+        return mWidgetDateFormat.format(calendar.getTime());
+    }
+
+    /**
+     * Convert Date from server format (yyyy-MM-dd) to Calendar
+     *
+     * @param serverDate Date from server
+     * @return Calendar with date from server
+     */
+    @NonNull
+    Calendar convertDateFromServerFormatToCalendar(@Nullable final String serverDate) {
+        final Calendar calendar = Calendar.getInstance();
+        if (isServerDateNotMatched(serverDate)) {
+            return calendar;
+        }
+
+        try {
+            calendar.setTime(mServerDateFormat.parse(serverDate));
+        } catch (ParseException e) {
+            Log.e(TAG, e.getMessage());
+        }
+        return calendar;
+    }
+
+    @NonNull
+    private Calendar getCalendar(final int year, final int month, final int dayOfMonth) {
+        final Calendar calendar = Calendar.getInstance();
+        calendar.set(year, month, dayOfMonth, 0, 0, 0);
+        return calendar;
+    }
+
+    private boolean isServerDateNotMatched(@Nullable final String serverDate) {
```

Comment:
please change this to `isServerDateValid` and update return to `return date != null;` 
or `isServerDateNotValid` and update return to `return date == null;`